kymatio/scattering1d/frontend/base_frontend.py

```diff
@@ -175,6 +253,42 @@ def output_size(self, detail=False):
             `'array'`, the output is a large array containing the
             concatenation of all scattering coefficients. Defaults to
             `'array'`.
+        pad_mode : str (default 'reflect') / function, optional
+            Name of padding scheme to use, one of (`x = [1, 2, 3]`):
+                - zero:    [0, 0, 1, 2, 3, 0, 0, 0]
+                - reflect: [3, 2, 1, 2, 3, 2, 1, 2]
+            Or, pad function with signature `pad_fn(x, pad_left, pad_right)`.
+            This sets `self.pad_mode='custom'` (the name of padding is used
+            for some internal logic).
+        max_pad_factor : int (default 2), optional
```

Comment:
This looks terrifying. Why would you ever want to pad a signal of size `256` by _twice_ its length on both sides?

Reply:
It's _max_ pad factor. And it's not absolute input length, but length of filters relative to input. I chose `2` because anything less will _severely_ degrade filters in the worst case of `J=log2_N`: we get pure sines.

Also, note that the whole x8 x16 padding thing isn't as scary as it seems: it _only_ applies for `J=log2_N`. That isn't a typical `J` at all, AFAIK. If `J=log2_N - 4`, we don't need to pad at all.

Reply:
Thanks for explaining. That makes sense.

You're doing a nice work making sure that everything functions as intended even in the most extreme cases (`J` approaching `log2_N`).

In practice, we will indeed have something like `J = log2_N - 4`, as you've pointed out.

I'll try to remember to explain this in the tutorial